WordPress/WordPressTest/Jetpack/JetpackInstallPromptTests.swift

```diff
@@ -0,0 +1,61 @@
+import XCTest
+@testable import WordPress
+
+class JetpackInstallPromptTests: XCTestCase {
+    private var contextManager: TestContextManager!
+    private var context: NSManagedObjectContext!
+    private var userDefaults: UserDefaults!
+    private var settings: JetpackInstallPromptSettings!
+
+    override func setUp() {
+        super.setUp()
+
+        contextManager = TestContextManager()
+        context = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        context.parent = contextManager.mainContext
+
+        let name = String(describing: JetpackInstallPromptSettings.self)
+        userDefaults = UserDefaults(suiteName: name)
+        userDefaults.removePersistentDomain(forName: name)
+
+        settings = JetpackInstallPromptSettings(userDefaults: userDefaults)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testPromptWillNotShowForDotComSites() {
+        let blog = ModelTestHelper.insertDotComBlog(context: context)
+        blog.dotComID = 1
+
+        XCTAssertFalse(settings.canDisplay(for: blog))
+    }
+
+    func testPromptWillShowForSitesWithoutJetpack() {
+        let blog = BlogBuilder(context).withJetpack(version: nil, username: nil, email: nil).build()
```

Comment:
[Q] Is `.withJetpack(version: nil, username: nil, email: nil)` equivalent to "no Jetpack"? Just wanted to make sure because the `.withJetpack(...)` confused me at first for tests without Jetpack.

Reply:
Good question! It's not the same since if you build a blog without using the `withJetpack` the jetpack blog options will not exist. When loading in the app normally this is always set with various bits of information, and the code I wrote depends on [that existing](https://github.com/wordpress-mobile/WordPress-iOS/pull/18319/files/c1d81d74ff51d0f6d898ea2a99e540afdf30c2b7#diff-cc7836f91bc8805ce71ffbfc5a00f4060a75c2c622bfc0964049afc461d53a66R23) or else it'll fail gracefully.